### PR TITLE
Preliminary Play 2.7 Support

### PR DESCRIPTION
**Pull request by arjunchhabra**
This is my first PR, so please let me know if there are any steps I missed. The project built successfully with a ./gradlew build.

Addresses issue #6362, here: https://github.com/gradle/gradle/issues/6362. 

All code is migrated directly from @raboof's PR, here: https://github.com/gradle/gradle/pull/6374.

There are three test files from raboof's PR that I was unable to locate in this repository:
PlayPlatformIntegrationTest.groovy
PlayPlatformResolverTest.groovy 
PlayDistributionPluginTest.groovy

Thanks,
Arjun

### CODE REVIEW

**Comment by arjunchhabra** [comment]
@lacasseio - Please take a look when you get a chance. Thanks!

**Comment by arjunchhabra** [comment]
Hmm, investigating the failure as ./gradlew integrationTest seems to complete successfully locally.

**Comment by arjunchhabra** [comment]
I think the integration tests are failing due to the same issues identified here: https://github.com/gradle/gradle/pull/6374#issuecomment-414365658 - but as I cannot seem to locate the three test files that were modified, I won't be able to address the integration tests. Am I understanding correctly?

**Comment by big-guy** [comment]
Thanks @arjunchhabra - hope to put out a 0.10 next week with these fixes.

File: src/integTestFixtures/groovy/org/gradle/playframework/fixtures/multiversion/PlayCoverage.groovy [hunk 1]
```diff
@@ -9,5 +9,5 @@ final class PlayCoverage {
     private PlayCoverage() {}
 
     public static final VersionNumber DEFAULT = VersionNumber.parse(DEFAULT_PLAY_VERSION)
-    public static final List<VersionNumber> ALL = [VersionNumber.parse('2.4.11'), VersionNumber.parse('2.5.18'), DEFAULT]
+    public static final List<VersionNumber> ALL = [VersionNumber.parse('2.4.11'), VersionNumber.parse('2.5.18'), DEFAULT, "2.7.0"]
```
Comment (big-guy) [inline src/integTestFixtures/groovy/org/gradle/playframework/fixtures/multiversion/PlayCoverage.groovy]: The problem you were seeing was that this expected a `VersionNumber` and you were passing a `String`

**Comment by arjunchhabra** [comment]
Hey @big-guy - please keep me updated whenever the release is cut. This is a blocking issue for us. Thanks!

**Comment by arjunchhabra** [comment]
> Thanks @arjunchhabra - hope to put out a 0.10 next week with these fixes.

@big-guy - Any update on this item?

**Comment by big-guy** [comment]
@arjunchhabra I'm caught up on getting the Gradle 6.0 release ready to go out, so I haven't come back around to this.

It looks like there are some failures that'll need to be figured out: https://scans.gradle.com/s/u75lsc43rkij4/tests/failed

Some of these are routes related and some of them are Twirl related. 

You could try locally with:
`./gradlew integrationTest --tests "*TwirlCompile*" -Pplayframework.int-test.target.version=2.7.3`

**Comment by decoded4620** [comment]
Appears to be a classpath issue, reflection trying to instantiate classes in unexpected (probably newer) formats.
```
Caused by: org.gradle.playframework.tools.internal.reflection.ObjectInstantiationException: Could not create an instance of type play.routes.compiler.RoutesCompiler$RoutesCompilerTask.
	at org.gradle.playframework.tools.internal.reflection.DirectInstantiator.newInstance(DirectInstantiator.java:35)...
org.gradle.playframework.tools.internal.routes.RoutesCompiler.compile(RoutesCompiler.java:53)
	... 16 more
Caused by: java.lang.IllegalArgumentException: Could not find any public constructor for class play.routes.compiler.RoutesCompiler$RoutesCompilerTask which accepts parameters [java.io.File, scala.collection.mutable.ListBuffer, java.lang.Boolean, java.lang.Boolean, java.lang.Boolean].
	at org.gradle.playframework.tools.internal.reflection.DirectInstantiator$ConstructorCache.create(DirectInstantiator.java:73)
...
```

**Pull request merged**